src/framework/standard/help_commands.rs

```diff
@@ -81,6 +82,95 @@ pub fn has_all_requirements(cmd: &Arc<CommandOptions>, msg: &Message) -> bool {
     !cmd.guild_only
 }
 
+fn analyse_command_for_suggestion(result: &mut Vec<String>, msg: &Message, command_options: &Arc<CommandOptions>, help_options: &HelpOptions, command_name: &str) {
+    if !command_options.dm_only && !command_options.guild_only
+    || command_options.dm_only && msg.is_private()
+    || command_options.guild_only && !msg.is_private() {
+
+        if let Some(guild) = msg.guild() {
+            let guild = guild.read();
+
+            if let Some(member) = guild.members.get(&msg.author.id) {
+
+                if command_options.help_available && has_correct_permissions(command_options, msg) {
+
+                    if has_correct_roles(command_options, &guild, &member) {
+                        result.push(format!("`{}`", command_name.clone()));
+                    } else {
+                        match help_options.lacking_role {
```

Comment:
These matches are unnecessarily repeated.
It'd be more appropriate to put this logic into a macro:
```rust
macro_rules! format_help {
    ($behaviour:expr) => {
        match $behaviour {
            HelpBehaviour::Strike => result.push(format!("~~`{}`~~", command_name)),
            HelpBehaviour::Nothing => result.push(format!("`{}`", command_name)),
            HelpBehaviour::Hide => {},
        }
    };
}
```
then it's plainly just
```rust
format_help!(help_options.lacking_role);
```
and so on.